h5netcdf/core.py

```diff
@@ -125,28 +132,55 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
+class _LazyObjectLookup(Mapping):
+    def __init__(self, parent, object_cls):
+        self._parent = parent
+        self._object_cls = object_cls
+        self._objects = OrderedDict()
+
+    def __setitem__(self, name, object):
```

Comment:
Another minor point: it's considered best practice to avoid shadowing built-ins like `object`. So I would probably better to rename to something else, e..g, `obj`.